api/controllers/v1/cave/update.js

```diff
@@ -13,37 +12,50 @@ const { toCave } = require('../../../services/mapping/converters');
 const { checkRight } = sails.helpers;
 
 module.exports = async (req, res) => {
-  // Check right
-  const hasRight = await checkRight
-    .with({
-      groups: req.token.groups,
-      rightEntity: RightService.RightEntities.CAVE,
-      rightAction: RightService.RightActions.EDIT_ANY,
-    })
-    .intercept('rightNotFound', () =>
-      res.serverError(
-        'A server error occured when checking your right to update a cave.'
-      )
-    );
-  if (!hasRight) {
-    return res.forbidden('You are not authorized to update a cave.');
-  }
+  try {
+    const hasRight = await checkRight
+      .with({
+        groups: req.token.groups,
+        rightEntity: RightService.RightEntities.CAVE,
+        rightAction: RightService.RightActions.EDIT_ANY,
+      })
+      .intercept('rightNotFound', () =>
+        res.serverError(
+          'A server error occured when checking your right to update a cave.'
+        )
+      );
+    if (!hasRight) {
+      return res.forbidden('You are not authorized to update a cave.');
+    }
```

Comment:
You don't need the `try` here: `intercept()` is syntaxic sugar for try/catch : https://sailsjs.com/documentation/reference/waterline-orm/queries/intercept

Reply:
Modified